dipy/viz/ui.py

```diff
@@ -0,0 +1,612 @@
+import numpy as np
+
+from dipy.viz.interactor import CustomInteractorStyle
+
+from dipy.utils.optpkg import optional_package
+
+# Allow import, but disable doctests if we don't have vtk.
+vtk, have_vtk, setup_module = optional_package('vtk')
+
+if have_vtk:
+    version = vtk.vtkVersion.GetVTKSourceVersion().split(' ')[-1]
+    major_version = vtk.vtkVersion.GetVTKMajorVersion()
+else:
+    vtkInteractorStyleUser = object
+
+
+class UI(object):
+    """ An umbrella class for all UI elements.
+    While adding UI elements to the renderer, we go over all the sub-elements
+    that come with it and add those to the renderer automatically.
+
+    Attributes
+    ----------
+    ui_param : object
+        This is an attribute that can be passed to the UI object by the interactor.
+    ui_list : list(UI)
```

Comment:
We use the numpy docstring standard: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt